kubeflow/pipelines.py

```diff
@@ -0,0 +1,445 @@
+            "python",
+            "-m",
+            "torch.distributed.launch",
+            f"--nproc_per_node={num_gpu}",
+            "datasetinsights",
+            "train",
+        ]
+    else:
+        command = ["datasetinsights", "train"]
+
+    arguments = [
+        f"--config={config}",
+        f"--train-data={train_data}",
+        f"--val-date={val_data}",
+        f"--tb_log_dir={tb_log_dir}",
+        f"--checkpoint-dir={checkpoint_dir}",
+    ]
+    if checkpoint_file:
+        arguments.append(f"--checkpoint-file={checkpoint_file}")
+
+    train = dsl.ContainerOp(
+        name="train",
+        image=docker,
+        command=command,
+        arguments=arguments,
+        pvolumes={DATA_PATH: volume},
+    )
+    # GPU
+    train.set_gpu_limit(num_gpu)
+    train.add_node_selector_constraint(
+        "cloud.google.com/gke-accelerator", gpu_type
+    )
+
+    train.set_memory_request(memory_limit)
+    train.set_memory_limit(memory_limit)
+
+    train.apply(gcp.use_gcp_secret("user-gcp-sa"))
+
+    return train
+
+
+def evaluate_op(
+    *,
+    docker,
+    config,
+    checkpoint_file,
+    test_data,
+    tb_log_dir,
+    volume,
+    memory_limit,
+    num_gpu,
+    gpu_type,
+):
+    """ Create a Kubeflow ContainerOp to evaluate an estimator.
+
+    Args:
+        docker (str): Docker image registry URI.
+        config (str): Path to estimator config file.
+        checkpoint_file (str): Path to an estimator checkpoint file.
+        test_data (str): Path to test dataset directory.
+        tb_log_dir (str): Path to tensorbload log directory.
+        checkpoint_dir (str): Path to checkpoint file directory.
+        volume (kfp.dsl.PipelineVolume): The volume where datasets are stored.
+        memory_limit (str): Set memory limit for this operator. For simplicity,
+            we set memory_request = memory_limit.
+        num_gpu (int): Set the number of GPU for this operator
+        gpu_type (str): Set the type of GPU
+
+    Returns:
+        kfp.dsl.ContainerOp: Represents an op implemented by a container image
+            to evaluate an estimator.
+    """
+    evaluate = dsl.ContainerOp(
+        name="evaluate",
+        image=docker,
+        command=["datasetinsights", "evaluate"],
+        arguments=[
+            f"--config={config}",
+            f"--checkpoint-file={checkpoint_file}",
+            f"--test-data={test_data}",
+            f"--tb_log_dir={tb_log_dir}",
+        ],
+        pvolumes={DATA_PATH: volume},
+    )
+    # GPU
+    evaluate.set_gpu_limit(num_gpu)
+    evaluate.add_node_selector_constraint(
+        "cloud.google.com/gke-accelerator", gpu_type
+    )
+
+    evaluate.set_memory_request(memory_limit)
+    evaluate.set_memory_limit(memory_limit)
+
+    evaluate.apply(gcp.use_gcp_secret("user-gcp-sa"))
+
+    return evaluate
+
+
+@dsl.pipeline(
+    name="Train on the SynthDet sample",
+    description="Train on the SynthDet sample",
+)
+def train_on_synthdet_sample(
```

Comment:
Is there a difference between this dsl.pipeline and the other pipelines besides the default arguments?

Reply:
They are not very different. We can probably make a few pipeline and share some of the components, but this might be over-engineering these pipelines. Using default arguments makes it easier to demonstrate different training/evaluation procedures.